pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -472,91 +472,124 @@ func ResolvePipelineRunTask(
 	getTaskRun resources.GetTaskRun,
 	getRun GetRun,
 	getCondition GetCondition,
-	task v1beta1.PipelineTask,
+	pipelineTask v1beta1.PipelineTask,
 	providedResources map[string]*resourcev1alpha1.PipelineResource,
 ) (*ResolvedPipelineRunTask, error) {
-
 	rprt := ResolvedPipelineRunTask{
-		PipelineTask: &task,
+		PipelineTask: &pipelineTask,
 	}
 	rprt.CustomTask = isCustomTask(ctx, rprt)
 	if rprt.IsCustomTask() {
-		rprt.RunName = getRunName(pipelineRun.Status.Runs, pipelineRun.Status.ChildReferences, task.Name, pipelineRun.Name)
+		rprt.RunName = getRunName(pipelineRun.Status.Runs, pipelineRun.Status.ChildReferences, pipelineTask.Name, pipelineRun.Name)
 		run, err := getRun(rprt.RunName)
 		if err != nil && !kerrors.IsNotFound(err) {
 			return nil, fmt.Errorf("error retrieving Run %s: %w", rprt.RunName, err)
 		}
 		rprt.Run = run
 	} else {
-		rprt.TaskRunName = GetTaskRunName(pipelineRun.Status.TaskRuns, pipelineRun.Status.ChildReferences, task.Name, pipelineRun.Name)
-
-		taskRun, err := getTaskRun(rprt.TaskRunName)
-		if err != nil {
-			if !kerrors.IsNotFound(err) {
-				return nil, fmt.Errorf("error retrieving TaskRun %s: %w", rprt.TaskRunName, err)
-			}
-		}
-		if taskRun != nil {
-			rprt.TaskRun = taskRun
+		rprt.TaskRunName = GetTaskRunName(pipelineRun.Status.TaskRuns, pipelineRun.Status.ChildReferences, pipelineTask.Name, pipelineRun.Name)
+		if err := resolvePipelineRunTaskWithTaskRun(ctx, &rprt, rprt.TaskRunName, pipelineRun, getTask, getTaskRun, getCondition, pipelineTask, providedResources); err != nil {
+			return nil, err
 		}
+	}
+	return &rprt, nil
+}
 
-		// Find the Task that this PipelineTask is using
-		spec, taskName, kind, err := resolveTask(ctx, taskRun, getTask, task)
-		if err != nil {
-			return nil, err
+func resolvePipelineRunTaskWithTaskRun(
+	ctx context.Context,
+	rprt *ResolvedPipelineRunTask,
+	taskRunName string,
+	pipelineRun v1beta1.PipelineRun,
+	getTask resources.GetTask,
+	getTaskRun resources.GetTaskRun,
+	getCondition GetCondition,
+	pipelineTask v1beta1.PipelineTask,
+	providedResources map[string]*resourcev1alpha1.PipelineResource,
+) error {
+	taskRun, err := getTaskRun(taskRunName)
+	if err != nil {
+		if !kerrors.IsNotFound(err) {
+			return fmt.Errorf("error retrieving TaskRun %s: %w", taskRunName, err)
 		}
+	}
+	rprt.TaskRun = taskRun
```

Comment:
if condition is missing here:

```
if taskRun != nil {
```

Reply:
I believe that the check was unnecessary because `rprt.TaskRun` is initialized as nil so if `taskRun` is nil, there's no need to check for that condition as it makes no difference - however, will add back the check just in case